autopilot/followpoint.cpp

```diff
@@ -70,64 +75,81 @@ void FollowPoint::holdPosition()
         mMovementController->setDesiredSteering(0.0);
         mMovementController->setDesiredSpeed(0.0);
     } else {
-        mVehicleConnection->requestVelocityAndYaw({}, mVehicleConnection->getVehicleState()->getPosition(mPosTypeUsed).getYaw());
+        mVehicleConnection->requestGotoENU(getCurrentVehiclePosition().getXYZ(), true);
     }
 }
 
-void FollowPoint::setPosTypeUsed(const PosType &posTypeUsed)
+void FollowPoint::pointToFollowInVehicleFrame(const PosPoint &point)
 {
-    mPosTypeUsed = posTypeUsed;
+    if (pointIsNew(point)) {
+        // ToDo: Apply follow point parameters here
+        mLineFromVehicleToPoint.setP1(QPointF(0,0));
+        mLineFromVehicleToPoint.setP2(point.getPoint());
+        mDistanceToPointIn2D = mLineFromVehicleToPoint.length();
+    }
 }
 
-void FollowPoint::updateFollowPointInVehicleFrame(const PosPoint &point)
+void FollowPoint::pointToFollowInEnuFrame(const PosPoint &point)
 {
+    if (pointIsNew(point)) {
+        // ToDo: Apply follow point parameters here
+        mCurrentPointToFollowInEnuFrame = point;
+        mCurrentPointToFollowInEnuFrame.setHeight(point.getHeight() + mFollowPointHeight);
+
+        mDistanceToPointIn2D = getCurrentVehiclePosition().getDistanceTo(mCurrentPointToFollowInEnuFrame);
+    }
+}
+
+bool FollowPoint::pointIsNew(const PosPoint &point)
+{
+    static QTime oldPointTime = QTime::currentTime().addSecs(-QDateTime::currentDateTime().offsetFromUtc());
+
     if ((mStmState == FollowPointSTMstates::FOLLOWING || mStmState == FollowPointSTMstates::WAITING) &&
-         (point.getTime() > mCurrentFollowPointInVehicleFrame.getTime())) {
-        if (mFollowPointTimedOut) {
+            (point.getTime() > oldPointTime)) {
+        if (mFollowPointTimedOut)
             qDebug() << "Follow Point: timeout reset.";
-        }
 
-        mCurrentFollowPointInVehicleFrame = point;
         mFollowPointHeartbeatTimer.start(mFollowPointTimeout_ms);
         mFollowPointTimedOut = false;
+        return true;
     }
+    return false;
 }
 
 PosPoint FollowPoint::getCurrentVehiclePosition()
```

Comment:
calling mVehicleState->getPosition(mPosTypeUsed); as such is probably clearer than introducing this function after the changes below.